packages/components/src/custom-select-control/types.ts

```diff
@@ -120,4 +120,13 @@ export type CustomSelectProps< T extends CustomSelectOption > = {
 	 * @default false
 	 */
 	__next40pxDefaultSize?: boolean;
+	/**
+	 * Enables legacy behavior for the custom select dropdown.
+	 *
+	 * When set to `true`, the dropdown will follow the legacy implementation, including
+	 * positioning and interaction styles. Set to `false` to use the updated behavior.
+	 *
+	 * @default true
+	 */
+	isLegacy?: boolean;
```

Comment:
This is not a prop intended to be exposed publicly, and is likely to cause unintended side effects or maintenance headaches when used in this way. I also looked into the `blame` for the flip logic and it seems like it was a pretty intentional hotfix for #63180. We do need to find a good fix for it though, to unblock CustomSelectControlV2. (cc @ciampo)

Reply:
(Also I think I'm open to setting `flip` to `true` for the legacy version as well, once the z-index issue is addressed.)

Reply:
Can't we expose flip behaviour publicly via props? I think It would make more sense to allow for it instead of changing anything in legacy.